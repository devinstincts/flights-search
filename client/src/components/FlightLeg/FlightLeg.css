.flight-leg {
    margin-bottom: 8px;
}
.flight-details-row {
    display: flex;
    align-items: center;
}
.flight-carrier-img {
    display: inline-block;
    width: 60px;
    height: 32px;
}
.flight-total-duration {
    justify-self: flex-end;
}
.flight-time-source-dest {
    flex: 1;
    min-Width: 200px;
    display: flex;
    align-items: center;
    font-size: 14px;
}
.flight-source-dest-details {
    margin-left: 20px;
}
.flight-leg-arrow-icon {
    height: 16px;
    display: inline-flex;
    align-items: center;
    margin: 0 8px 0 28px;
}
.flight-time {
    font-weight: bold;
}
.flight-code {
    color: #4c4c4c;
    opacity: 0.8;
    font-weight: 500;
}
.fight-total-time {
    color: #4c4c4c;
    font-size: 12px;
    opacity: 0.8;
}
.flight-direct {
    color: #00ca6e;
    font-weight: bold;
    font-size: 13px;
}
