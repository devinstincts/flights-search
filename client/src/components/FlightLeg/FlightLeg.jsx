import React from 'react';
import STYLES from './FlightLeg.scss';
import rightArrowIcon from './right-arrow.svg';
import util from '../../util';
import AppPropTypes from '../../prop-types';

const c = (className) => STYLES[className] || 'UNKNOWN';

const BLOCK_NAME = 'FlightLeg';

const stopsTextClass = (value) => (value === 'Direct' ? c(`${BLOCK_NAME}__duration-details__stops_text`) : '');

const FlightLeg = ({ flightLeg }) => {
  const departure = new Date(`${flightLeg.Departure}Z`);
  const arrival = new Date(`${flightLeg.Arrival}Z`);
  const stopText = util.stopsText(flightLeg.Stops);
  return (
    <div className={c(BLOCK_NAME)}>
      <div className={c(`${BLOCK_NAME}__OND-container`)}>
        <span className={c(`${BLOCK_NAME}__OND-icon`)}>
          <img
            className={c(`${BLOCK_NAME}__OND-carrier-image`)}
            alt={flightLeg.Carriers[0].Code}
            src={flightLeg.Carriers[0].ImageUrl}
          />
        </span>
        <div className={c(`${BLOCK_NAME}__origin-details`)}>
          <p>
            {`${departure.toLocaleTimeString().substr(0, 5)}`}
          </p>
          <p>{flightLeg.OriginStation.Code}</p>
        </div>
        <img className={c(`${BLOCK_NAME}__OND-right-icon`)} alt="To" src={rightArrowIcon} />
        <div className={c(`${BLOCK_NAME}__destination-details`)}>
          <p>
            {`${arrival.toLocaleTimeString().substr(0, 5)}`}
          </p>
          <p>{flightLeg.DestinationStation.Code}</p>
        </div>
      </div>
      <div className={c(`${BLOCK_NAME}__duration-details`)}>
        <p>{util.minutesInHrsAndMins(flightLeg.Duration)}</p>
        <p className={stopsTextClass(stopText)}>{stopText}</p>
      </div>
    </div>
  );
};

FlightLeg.propTypes = {
  flightLeg: AppPropTypes.Leg.isRequired,
};

export default FlightLeg;
