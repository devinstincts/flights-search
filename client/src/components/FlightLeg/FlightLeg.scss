@import '~bpk-mixins';

.FlightLeg {
    margin: 0 0 15px 0;
    background: #fff;
    display: flex;
    justify-content: space-between;

    &__parent-container {
        background: #fff;
        border-radius: 10px;
        padding: 10px;
        margin: 15px 0px 15px 0px;
    }

    &__origin-details, 
    &__destination-details {
        p {
            margin: 0;
            font-size: 14px;
            line-height: normal;

            &:first-child {
                margin: 0 0 10px;
            }
        }
    }

    &__origin-details {
        margin-right: 15px;
    }

    &__duration-details {
        p {
            font-size: 12px;
            margin: 0;
            line-height: normal;

            &:first-child {
                margin: 0 0 10px;
            }
        }

        &__stops_text {
            color: #00ca6e;
            font-weight: bold;
        }
    }

    &__OND {
        &-container {
            display: flex;
            flex-direction: row;
        }
        &-icon {
            margin: 5px 15px 0 0;
        }
        &-right-icon {
            margin-right: 15px; 
        }
        &-carrier-image {
            width: 60px;
            height: 30px;
        }
    }

    &__footer-details{
        display: flex;
        justify-content: space-between;
        align-items: flex-end;

        &-amount {
            p {
                margin: 0;
                line-height: normal;
                &:first-child {
                    margin-bottom: 5px;
                    font-size: 20px;
                }
            }
        }

        button {
            padding: 6px 31px;
            background: #00ca6e;
            border-radius: 25px;
            color: #fff;
            border: 0;
            font-weight: bold;
            height: 35px;
            font-size: 14px;
        }
    }
}