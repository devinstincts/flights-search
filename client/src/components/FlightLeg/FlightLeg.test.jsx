import React from 'react';
import ReactDOM from 'react-dom';
import ShallowRenderer from 'react-test-renderer/shallow';
import FlightLeg from './index';
import mockRes from '../../mock.json';

describe('FlightLeg', () => {
  it('should render without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<FlightLeg flightLeg={mockRes.Itineraries[0].inboundLeg} />, div);
  });

  it('should render correctly', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<FlightLeg flightLeg={mockRes.Itineraries[0].inboundLeg} />);
    // expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
