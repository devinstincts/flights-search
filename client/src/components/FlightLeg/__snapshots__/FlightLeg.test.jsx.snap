// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlightLeg should render correctly 1`] = `
<div
  className="UNKNOWN"
>
  <div
    className="UNKNOWN"
  >
    <span
      className="UNKNOWN"
    >
      <img
        alt="BA"
        className="UNKNOWN"
        src="https://s1.apideeplink.com/images/airlines/BA.png"
      />
    </span>
    <div
      className="UNKNOWN"
    >
      <p>
        11:05
      </p>
      <p>
        LHR
      </p>
    </div>
    <img
      alt="To"
      className="UNKNOWN"
      src="right-arrow.svg"
    />
    <div
      className="UNKNOWN"
    >
      <p>
        12:30
      </p>
      <p>
        EDI
      </p>
    </div>
  </div>
  <div
    className="UNKNOWN"
  >
    <p>
      1h 25m
    </p>
    <p
      className="UNKNOWN"
    >
      Direct
    </p>
  </div>
</div>
`;
